tests/test_vqe.py

```diff
@@ -1212,6 +1212,145 @@ def test_variance_error(self):
             qml.ExpvalCost(qml.templates.StronglyEntanglingLayers, hamiltonian, dev, measure="var")
 
 
+class TestNewVQE:
```

Comment:
What happens when `H` in `qml.expval(H)` is not a PL Hamiltonian? Maybe there should be a test for that?

Reply:
Excellent idea!

Reply:
Yep, I would say this is more in scope of `expval`